dotCMS/src/main/java/com/liferay/portal/servlet/AngularRoutingFilter.java

```diff
@@ -0,0 +1,58 @@
+package com.liferay.portal.servlet;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * If the request is a Angular routing request then it response with a 304 Http code.
+ */
+public class AngularRoutingFilter implements Filter {
+
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+
+        String path = ((HttpServletRequest) servletRequest).getServletPath();
+
+        if (!path.toLowerCase().endsWith("js")
```

Comment:
This shouldn't be an array coming from the configuration with a default list?
Or might be an static collection, I think you can apply a lambda or something more elegante to avoid the if